pkg/cloud/ovirt/providerIDcontroller/providerIDController.go

```diff
@@ -0,0 +1,159 @@
+package providerIDcontroller
+
+import (
+	"context"
+	"fmt"
+
+	ovirtsdk "github.com/ovirt/go-ovirt"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/openshift/cluster-api-provider-ovirt/pkg/cloud/ovirt/clients"
+)
+
+var _ reconcile.Reconciler = &ProviderIDReconciler{}
+
+type ProviderIDReconciler struct {
+	operatorName         string
+	client               client.Client
+	listNodesByFieldFunc func(key, value string) ([]corev1.Node, error)
+	fetchProviderIDFunc  func(string) (string, error)
+	ovirtApi             *ovirtsdk.Connection
+}
+
+func (p *ProviderIDReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	klog.Infof("Reconciling providerID for node %s", request.NamespacedName)
+
+	// Fetch the Node instance
+	node := corev1.Node{}
+	err := p.client.Get(context.Background(), request.NamespacedName, &node)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		klog.Errorf("Error getting node: %v", err)
+		return reconcile.Result{}, fmt.Errorf("error getting node: %v", err)
+	}
+
+	if node.Spec.ProviderID != "" {
```

Comment:
might want to reconcile against expected value here.

Reply:
what do you mean?

Reply:
Eg, if someone manually changed this, your controller should put it back to whatever is the right or correct value, you could achieve this by always fetching the ID from oVirt, comparing with the existing value, and only updating if they differ

Reply:
editing a node.providerID is blocked by validation. once set it can't be changed. That's why it only makes sense to detect empty values.

Reply:
If that's true then fair enough, no changes needed here 😄 

I assume this is built into the API server? Might be worth a validation of this as a test in the test suite (could use the envtest from controller-runtime) to verify this and make sure that behaviour doesn't change in the future, or if it does, signal that you need to update your logic